debezium-server/debezium-server-core/src/main/java/io/debezium/server/DebeziumMetrics.java

```diff
@@ -0,0 +1,130 @@
+    public static final MBeanServer mbeanServer = ManagementFactory.getPlatformMBeanServer();
+    @ConfigProperty(name = "debezium.source.max.queue.size", defaultValue = CommonConnectorConfig.DEFAULT_MAX_QUEUE_SIZE + "")
+    int maxQueueSize;
+    private ObjectName snapshotMetricsObjectName;
+    private ObjectName streamingMetricsObjectName;
+
+    private static ObjectName getDebeziumMbean(String context) {
+        ObjectName debeziumMbean = null;
+
+        for (ObjectName mbean : mbeanServer.queryNames(null, null)) {
+
+            if (mbean.getCanonicalName().contains("debezium.")
+                    && mbean.getCanonicalName().contains("type=connector-metrics")
+                    && mbean.getCanonicalName().contains("context=" + context)) {
+                LOGGER.debug("Using {} MBean to get {} metrics", mbean, context);
+                debeziumMbean = mbean;
+                break;
+            }
+
+        }
+
+        Objects.requireNonNull(debeziumMbean, "Debezium MBean (context=" + context + ") not found!");
+
+        return debeziumMbean;
+    }
+
+    public ObjectName getSnapshotMetricsObjectName() {
+
+        if (snapshotMetricsObjectName == null) {
+            snapshotMetricsObjectName = getDebeziumMbean("snapshot");
+        }
+
+        return snapshotMetricsObjectName;
+    }
+
+    public ObjectName getStreamingMetricsObjectName() {
+
+        if (streamingMetricsObjectName == null) {
+            streamingMetricsObjectName = getDebeziumMbean("streaming");
+        }
+
+        return streamingMetricsObjectName;
+    }
+
+    public int maxQueueSize() {
+        return maxQueueSize;
+    }
+
+    public boolean snapshotRunning() {
+        try {
+            return (boolean) mbeanServer.getAttribute(getSnapshotMetricsObjectName(), "SnapshotRunning");
+        }
+        catch (Exception e) {
+            throw new DebeziumException(e);
+        }
+    }
+
+    public boolean snapshotCompleted() {
+        try {
+            return (boolean) mbeanServer.getAttribute(getSnapshotMetricsObjectName(), "SnapshotCompleted");
+        }
+        catch (Exception e) {
+            throw new DebeziumException(e);
+        }
+    }
+
+    public int streamingQueueRemainingCapacity() {
+        try {
+            return (int) mbeanServer.getAttribute(getStreamingMetricsObjectName(), "QueueRemainingCapacity");
+        }
+        catch (Exception e) {
+            throw new DebeziumException(e);
+        }
+    }
+
+    public int streamingQueueCurrentSize() {
+        return maxQueueSize - streamingQueueRemainingCapacity();
+    }
+
+    public long streamingMilliSecondsBehindSource() {
+        try {
+            return (long) mbeanServer.getAttribute(getStreamingMetricsObjectName(), "MilliSecondsBehindSource");
+        }
+        catch (Exception e) {
+            throw new DebeziumException(e);
+        }
+    }
+
+    public void logMetrics() {
+        LOGGER.info("Debezium Metrics:" +
```

Comment:
Could you please use `{}` placeholders in logging message instead of constructing the log message as a string?

Reply:
done